File: libs/document-processor-next/src/constants/processor-pointer-constants.ts
/**
 * 1:1 port of {@code blue.language.processor.util.ProcessorPointerConstants}.
 */
import {
  KEY_CHECKPOINT,
  KEY_EMBEDDED,
  KEY_INITIALIZED,
  KEY_TERMINATED,
} from './processor-contract-constants.js';

export const RELATIVE_CONTRACTS = '/contracts' as const;
export const RELATIVE_INITIALIZED = `${RELATIVE_CONTRACTS}/${KEY_INITIALIZED}` as const;
export const RELATIVE_TERMINATED = `${RELATIVE_CONTRACTS}/${KEY_TERMINATED}` as const;
export const RELATIVE_EMBEDDED = `${RELATIVE_CONTRACTS}/${KEY_EMBEDDED}` as const;
export const RELATIVE_CHECKPOINT = `${RELATIVE_CONTRACTS}/${KEY_CHECKPOINT}` as const;

const LAST_EVENTS_SUFFIX = '/lastEvents';
const LAST_SIGNATURES_SUFFIX = '/lastSignatures';

export function relativeContractsEntry(key: string): string {
  return `${RELATIVE_CONTRACTS}/${key}`;
}

export function relativeCheckpointLastEvent(
  markerKey: string,
  channelKey: string
): string {
  return `${relativeContractsEntry(markerKey)}${LAST_EVENTS_SUFFIX}/${channelKey}`;
}

export function relativeCheckpointLastSignature(
  markerKey: string,
  channelKey: string
): string {
  return `${relativeContractsEntry(markerKey)}${LAST_SIGNATURES_SUFFIX}/${channelKey}`;
}

File: references/java-sdk/src/main/java/blue/language/processor/util/ProcessorPointerConstants.java
package blue.language.processor.util;

/**
 * Shared relative pointer constants for processor-managed contract paths.
 *
 * <p>Centralises the JSON-pointer fragments the runtime relies on when reading or
 * writing reserved contract entries. Keeping them here avoids drift between
 * runtime logic, tests, and documentation.</p>
 */
public final class ProcessorPointerConstants {

    public static final String RELATIVE_CONTRACTS = "/contracts";
    public static final String RELATIVE_INITIALIZED = RELATIVE_CONTRACTS + "/" + ProcessorContractConstants.KEY_INITIALIZED;
    public static final String RELATIVE_TERMINATED = RELATIVE_CONTRACTS + "/" + ProcessorContractConstants.KEY_TERMINATED;
    public static final String RELATIVE_EMBEDDED = RELATIVE_CONTRACTS + "/" + ProcessorContractConstants.KEY_EMBEDDED;
    public static final String RELATIVE_CHECKPOINT = RELATIVE_CONTRACTS + "/" + ProcessorContractConstants.KEY_CHECKPOINT;

    private static final String LAST_EVENTS_SUFFIX = "/lastEvents";
    private static final String LAST_SIGNATURES_SUFFIX = "/lastSignatures";

    private ProcessorPointerConstants() {
    }

    public static String relativeContractsEntry(String key) {
        return RELATIVE_CONTRACTS + "/" + PointerUtils.escapeRequiredPointerSegment(key, "Contract key");
    }

    public static String relativeCheckpointLastEvent(String markerKey, String channelKey) {
        return relativeContractsEntry(markerKey) + LAST_EVENTS_SUFFIX + "/"
                + PointerUtils.escapeRequiredPointerSegment(channelKey, "Channel key");
    }

    public static String relativeCheckpointLastSignature(String markerKey, String channelKey) {
        return relativeContractsEntry(markerKey) + LAST_SIGNATURES_SUFFIX + "/"
                + PointerUtils.escapeRequiredPointerSegment(channelKey, "Channel key");
    }
}
